etcdserver/v3demo_server.go

```diff
@@ -398,19 +422,12 @@ func applyTxn(kv dstorage.KV, le lease.Lessor, rt *pb.TxnRequest) (*pb.TxnRespon
 		}
 	}
 
-	// TODO: check potential errors before actually applying anything
-
-	var reqs []*pb.RequestUnion
 	if ok {
 		reqs = rt.Success
 	} else {
 		reqs = rt.Failure
 	}
 
-	if err := checkRequestLeases(le, reqs); err != nil {
```

Comment:
I'm leaning toward the permissive model of accepting bogus requests if they never execute; more sorts of transactions would go through.

Two cases where it might make sense to do permissive:
* `if (key.modify == foreverAgo) { rangerequest [begin, end) foreverAgo } else { rangerequest [begin, end) mostRecent} ` some sort of archiving maybe?
* `if whatever { put with lease1 } else { put with lease2}` where lease1 and lease2 have unrelated expirations (probably useful, not sure for what)